nixos/tests/bpf.nix

```diff
@@ -0,0 +1,25 @@
+import ./make-test-python.nix ({ pkgs, ... }: {
+  name = "bpf";
+  meta.maintainers = with pkgs.lib.maintainers; [ martinetd ];
+
+  machine = { pkgs, ... }: {
+    programs.bcc.enable = true;
+    environment.systemPackages = with pkgs; [ bpftrace ];
+  };
+
+  testScript = ''
+    ## bcc
+    # syscount -d 1 stops 1s after probe started so is good for that
+    print(machine.succeed("syscount -d 1"))
+
+    ## bpftrace
+    # list probes
+    machine.succeed("bpftrace -l")
```

Comment:
This test can be added to `passthru.tests` of the bpftrace package.

Reply:
I just found out about passthru.tests earlier today when trying to get OfBorg to run tests for me -- I think that's a great idea.
We do need this kind of tests for programs.bcc.enable though so should I just leave the bcc one there, or is the module simple enough (it's really just extra packages...) that this could go to bcc's passthru.tests instead as well? If that's easier to get coverage I'd be for moving it.

I'll do this as soon as I figure bcc's build first and check the exact syntax. Gotta afk a bit, might be a while - marking this PR draft for now.

Reply:
I think I had misunderstood this comment - I wanted to try to move the commands as runCommand in bpftrace's passthru.test, but these don't run as root so we need to keep these here.

I've added nixosTests.bpf to bpftrace and bcc's passthru.test instead so now if either change this should automatically run. I assume that's what you had intended in the first place?

Reply:
yes.